sensor/common/scannerclient/grpc_client.go

```diff
@@ -2,72 +2,172 @@ package scannerclient
 
 import (
 	"context"
+	"fmt"
 	"strings"
 
 	"github.com/pkg/errors"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/env"
 	"github.com/stackrox/rox/pkg/images/utils"
 	"github.com/stackrox/rox/pkg/logging"
 	"github.com/stackrox/rox/pkg/mtls"
 	"github.com/stackrox/rox/pkg/registries/types"
 	scannerV1 "github.com/stackrox/scanner/generated/scanner/api/v1"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
 )
 
 var (
 	log = logging.LoggerForModule()
 )
 
-// Client is a Scanner gRPC Client.
-type Client struct {
-	client scannerV1.ImageScanServiceClient
-	conn   *grpc.ClientConn
+// ScannerClient is the interface to remote image scanners used by Sensor.
+type ScannerClient interface {
+	GetImageAnalysis(context.Context, *storage.Image, *types.Config) (*ImageAnalysis, error)
+	Close() error
 }
 
-// dial Scanner and return a new Client.
-// dial is non-blocking and returns a non-nil error upon configuration error.
-func dial(endpoint string) (*Client, error) {
-	if endpoint == "" {
-		return nil, errors.New("Invalid Scanner endpoint (empty)")
+// ImageAnalysis is the result of an image analysis.
+type ImageAnalysis struct {
+	ScanStatus scannerV1.ScanStatus
+	ScanNotes  []scannerV1.Note
+	// Fields for analysis results for each supported Scanner: Scanner V2 (v1 proto)
+	// and Scanner V4 (v4 proto).
+	V1Components *scannerV1.Components
+	V4Contents   *v4.Contents
+}
+
+// v2Client is the client for StackRox Scanner based on Clair V2, also
+// known as Scanner V2.
+type v2Client struct {
```

Comment:
I know in other places I suggested nameVersion, but I think the way this is done now makes sense and no need to change (v2Client vs potentially changing to clientV2)